fineract-core/src/main/java/org/apache/fineract/commands/service/SynchronousCommandProcessingService.java

```diff
@@ -272,6 +272,11 @@ private void publishHookEvent(final String entityName, final String actionName,
                 Type type = new TypeToken<Map<String, Object>>() {
 
                 }.getType();
+
+                if (!isValidJson(command.json(), type)) {
```

Comment:
This method is not required. I would just simply wrap the `Map<String, Object> myMap = gson.fromJson(command.json(), type);` around with try catch and throw the `PlatformApiDataValidationException` exception.

Reply:
This way we can avoid double json conversion